aws/src/main/java/org/apache/iceberg/aws/AwsClientFactories.java

```diff
@@ -226,7 +226,7 @@ static AwsCredentialsProvider credentialsProvider(
             AwsSessionCredentials.create(accessKeyId, secretAccessKey, sessionToken));
       }
     } else {
-      return DefaultCredentialsProvider.create();
+      return DefaultCredentialsProvider.builder().build();
```

Comment:
Nit: Maybe have an inline comment along the lines of "Create a new credential provider for each client"

Reply:
Comment added as requested.

Reply:
@Kontinuation if we add the comment, let's add it to all the places

Reply:
done